### ISSUE TITLE
[a11y]: Coachmark Beacon Uses role="tooltip"

### ISSUE THREAD

**Issue by VineethaV24-01** [carbon-design-system/ibm-products#7935]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.64.0

### React version

18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The beacon includes a <span> with role="tooltip" that wraps a button element.

<img width="720" height="430" alt="Image" src="https://github.com/user-attachments/assets/fc8be4db-11ff-4ba7-8d79-c65fb9b05c1c" />

In my application, a Coachmark beacon above a button is used to guide user focus during a UI tour. The beacon includes a <span> with role="tooltip" that wraps a button element. 
This setup triggers the following accessibility violation: 

<img width="1092" height="126" alt="Image" src="https://github.com/user-attachments/assets/4cdea793-fb3f-4c23-9636-3457c749af78" />

It seems the violation is due to the tooltip role being applied to a wrapper element around the button.
**Looking for clarification on:**
Why role="tooltip" is used in this context.?
Whether it's necessary or can be adjusted to avoid the violation.?

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/docs/experimental-onboarding-coachmark-coachmark--overview

### Steps to reproduce

Check the element in  browser inspect mode we can see that the beacon includes a <span> with role="tooltip" that wraps a button element. 

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
